code/modules/antagonists/heretic/knowledge/side_void_blade.dm

```diff
@@ -0,0 +1,177 @@
+// Sidepaths for knowledge between Void and Blade.
+
+/// The max health given to Shattered Risen
+#define RISEN_MAX_HEALTH 125
+
+/datum/heretic_knowledge/limited_amount/risen_corpse
+	name = "Shattered Ritual"
+	desc = "Allows you to transmute a corpse with a soul, a pair of latex or nitrile gloves, and \
+		and any exosuit clothing (such as armor) to create a Shattered Risen. \
+		Shattered Risen are strong ghouls that have 125 health, but cannot hold items, \
+		instead having two brutal weapons for hands. You can only create one at a time."
+	gain_text = "I witnessed a cold, rending force drag this corpse back to near-life. \
+		When it moves, it crunches like broken glass. Its hands are no longer recognizable as human - \
+		each clenched fist contains a brutal nest of sharp bone-shards instead."
+	next_knowledge = list(
+		/datum/heretic_knowledge/cold_snap,
+		/datum/heretic_knowledge/blade_dance,
+	)
+	required_atoms = list(
+		/obj/item/clothing/suit = 1,
+		/obj/item/clothing/gloves/color/latex = 1,
+		/mob/living/carbon/human = 1,
+	)
+	limit = 1
+	cost = 1
+	route = PATH_SIDE
+
+/datum/heretic_knowledge/limited_amount/risen_corpse/recipe_snowflake_check(mob/living/user, list/atoms, list/selected_atoms, turf/loc)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	for(var/mob/living/carbon/human/body in atoms)
+		if(body.stat != DEAD || !IS_VALID_GHOUL_MOB(body) || HAS_TRAIT(body, TRAIT_HUSK))
+			atoms -= body
+			continue
+
+		// We will only grab bodies with minds that have ghosts (with clients)
+		if(!body.mind?.get_ghost(ghosts_with_clients = TRUE))
+			atoms -= body
+			continue
+
+	if(!(locate(/mob/living/carbon/human) in atoms))
+		loc.balloon_alert(user, "ritual failed, no valid body!")
+		return FALSE
```

Comment:
maybe just handle this via variable instead. Break early when you've found your target body

Reply:
That kinda thing isn't very supported by heretic rituals in their current state, but I changed it round a bit anyways